### PR TITLE
[SCHAINS-242] Remove bitcore node dependency

### PR DESCRIPTION
**Pull request by otoumas**
In this PR the needed changes to use the node package of bitcore-module-sidechain are done. The changes are:
- Dependency: npm package instead of git repository
- Changes in import to adjust to new package
- Changes in Dockerfile to adjust to new package

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,4 @@ package-lock.json
 bitcore.config.json
```
Comment (cronicc) [inline .gitignore:8]: Please commit package-lock.json

File: Dockerfile [hunk 2]
```diff
@@ -1,17 +1,20 @@
...
 COPY package.json /sc_audit_explorer/
 COPY tsconfig.json /sc_audit_explorer/
 
 RUN set -euxo pipefail \
-    && git config --global url."https://${GITHUB_TOKEN}@github.com/".insteadOf "https://github.com/" \
+    && touch .npmrc \
+    && echo //npm.pkg.github.com/:_authToken=${GITHUB_TOKEN} >> .npmrc \
+    && echo  @horizenofficial:registry=https://npm.pkg.github.com >> .npmrc \
     && npm install \
     && npm run compile
```
Comment (cronicc) [inline Dockerfile:19]: Unless our nodejs code can handle signals (SIGINT, SIGTERM) and reap zombie processes an init system should be used like https://github.com/krallin/tini.

See: https://developpaper.com/avoid-running-nodejs-as-pid-1-under-the-docker-image/

File: package.json [hunk 3]
```diff
@@ -3,52 +3,48 @@
...
     "clean": "rm -rf build",
     "tsc": "npm run clean && tsc",
-    "start": "export BITCORE_CONFIG_PATH=$(pwd)/bitcore.config.json  && node build/node_modules/bitcore-node/src/server.js",
+    "start": "export BITCORE_CONFIG_PATH=$(pwd)/bitcore.config.json  && node node_modules/@horizenofficial/bitcore-module-sidechain/build/src/server.js",
     "dev": "npm run compile && npm run start",
-    "postinstall": "cd node_modules/bitcore-node && npm run compile",
     "test": "npm run compile && ./node_modules/mocha/bin/mocha -r ts-node/register test/**/*.ts"
   },
   "author": "",
   "license": "ISC",
```
Comment (cronicc) [inline package.json:14]:
```
  "engines": {
    "node": ">=16"
  },
```

File: tsconfig.json [hunk 4]
```diff
@@ -1,11 +1,13 @@
 {
   "compilerOptions": {
-    "target": "ES2017",
+    "target": "ES2019",
```
Comment (cronicc) [inline tsconfig.json:3]: If node 16 is required, ES2021 could be used, see https://stackoverflow.com/questions/67371787/what-typescript-configuration-produces-output-closest-to-node-js-16-capabilities/67371788#67371788

**Review by cronicc — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -3,7 +3,7 @@
 ## Deployment
 #### Prerequisites
 - Check that you have permissions for these three repos: `bitcore-module-sidechain`, `sc-bitcore-node-zen`
```
Comment (MarcoOl94) [inline README.md:5]: I think sc-bitcore-node-zen was merged into bitcore-module-sidechain, if it's true remove it from the README please.

**Review by paolocappelletti — CHANGES_REQUESTED** [review]

**Review by cronicc — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -14,12 +14,13 @@
...
+- Self-contained node running
 #### Configuration
-- Copy the `bitcore.config.json.template` file and rename it to `bitcore.config.json`
-- Change the `path_to_repo` in the `modules` field
-- Change the `websocket.address` according to the environment you want to run the project
-- Configure your ssh key or your personal token: `git config --global url."https://${GITHUB_TOKEN}@github.com/".insteadOf "https://github.com/"`
+- Copy the `.npmrc.template` file and rename it to `.npmrc`
+- Change `GITHUB_TOKEN` by the token generated in the Prerequisites step
 ### How to run it
 - `npm install`
```
Comment (cronicc) [inline README.md:24]:
```suggestion
- `npm ci`
```

**Pull request merged**